src/XrdCrypto/XrdCryptosslgsiAux.cc

```diff
@@ -656,17 +656,61 @@ int XrdCryptosslX509CreateProxyReq(XrdCryptoX509 *xcpi,
       PRINT("EEC certificate has expired");
       return -kErrPX_ExpiredEEC;
    }
+
+   // These will be assigned dynamically allocated ssl structures later.
+   // An RAII style guard is used to free them on exit from this method.
+   EVP_PKEY       *ekro  = 0;
+   X509_EXTENSION *ext   = 0;
+   X509_NAME      *psubj = 0;
+   X509_REQ       *xro   = 0;
+   PROXY_CERT_INFO_EXTENSION *pci = 0;
+   STACK_OF(X509_EXTENSION)  *esk = 0;
+
+   class sslGuard {
+   public:
+     sslGuard(EVP_PKEY *&epk, X509_EXTENSION *&xext, X509_NAME *&xname,
+              X509_REQ *&xreq, PROXY_CERT_INFO_EXTENSION *&pci,
+              STACK_OF(X509_EXTENSION) *&ske) :
+            epk_(epk), xext_(xext), xname_(xname), xreq_(xreq), pci_(pci),
+            ske_(ske) { }
+
+       ~sslGuard() {
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+         sk_X509_EXTENSION_pop_free(ske_, X509_EXTENSION_free);
+#else /* OPENSSL */
```

Comment:
Can we drop support for pre-OpenSSL 1.0?  That would mean we drop support for RHEL5 (which seems OK).